Marble Blast Platinum/platinum/client/ui/ExitGameDlg.gui

```diff
@@ -302,30 +545,157 @@ function ExitGameDlg::onWake(%this) {
 		%awesomeTitle = "<shadow:1:1><shadowcolor:000000bf><color:FF4444>Awesome";
 	}
 
-	%parText      = %parTitle;
-	%goldText     = %goldTitle;
-	%platinumText = %platinumTitle;
-	%ultimateText = %ultimateTitle;
-	%awesomeText  = %awesomeTitle;
+	%recordTitle   = "<shadow:1:1><shadowcolor:000000bf><color:0060f0>Record";
+	%eggRecordTitle = "<shadow:1:1><shadowcolor:000000bf><color:4580ff>Egg Record";
+
+	//Get the world record
+	%record = false;
+	if (lb()) {
+		
+		%cache = PlayMissionGui.globalScoreCache[PlayMissionGui.getMissionInfo().id];
+		if (isObject(%cache)) {
+			%scores = %cache.scores;
+			if (%scores.getSize()) {
+				%record = %scores.getEntry(0).score;
+				%recordType = %scores.getEntry(0).score_type;
+				%recordLabel = (%recordType $= "time" ? formatTime(%record) : formatScore(%record));
+			}
+		}
 
-	//Par can have both time and score
-	if (ClientMode::callbackForMission(%info, "timeMultiplier", 1) > 0) {
-		if (%parTimeLabel !$= "N/A") {
-			%text = %text NL "<just:left><spush>" @ %parText @ ":<spop><just:right>" @ %parTimeLabel;
+		%cache = PlayMissionGui.globalScoreModeCache[PlayMissionGui.getMissionInfo().id]; 
+		if (isObject(%cache)) {
+			%scores = %cache.getFieldValue("egg").scores; // Egg scores
+			if (%scores.getSize()) {
+				%eggRecordLabel = formatTime(%scores.getEntry(0).time);
+			}
 		}
 	}
-	if (%parScoreLabel !$= "N/A") {
-		%text = %text NL "<just:left><spush>" @ %parTitle SPC "Score:<spop><just:right>" @ %parScoreLabel;
+	%challengeTitles = "";
+	%challengeTimes = "";
+
+	%info = PlayMissionGui.getMissionInfo();
+	%pbType = getField($hs[0], 0);
+	%pbLabel = %pbType == 0 ? formatTime(getField($hs[0], 1)) : formatScore(getField($hs[0], 1));
+	%pbColor = getScoreFormatting($hs[0], %info, "", (%record == getField($hs[0], 1))? 1 : 2); // Technically no record (offline) and score of 0 counts as a record, but PB should not show in that case.
```

Comment:
I worry that this will mark 0-point hunt scores as records offline. Would be better to have `%isRecord = false;` above the wr check and set it to true if you beat/matched the record. Will also need to check less-than for time, greater-than for score. Or if the two are different types, time always beats score.

Reply:
Actually you're right. It is possible to get a PB of 0 - but it won't show up due to the PB check. I guess it should, though, how should I check that? It says:
`%default = ClientMode::callbackForMission(%info, "getDefaultScore", $ScoreType::Time TAB 5999999 TAB "Matan W.");`

And looking at it, I'm not sure if it is going to detect a 0 score, looks like a 99:59:99 time. The more hacky way is checking "Matan W." username though.

Reply:
![unknown4](https://user-images.githubusercontent.com/43479839/110766037-c32df500-8222-11eb-8730-2e4d2a826f86.png)

> Will also need to check less-than for time, greater-than for score. Or if the two are different types, time always beats score.
The `getScoreFormatting` function is already doing all the work here with the colors. I'm not sure what this means, I haven't had it look strange at all for me, except maybe that it doesn't show the light-red that appears on some UI elements if you fail par, which is just a small subjective thing.